### PR TITLE
Fix problems with Gating Grid simulation, add fix from Xianglei for i…

### PR DESCRIPTION
**Pull request by fisyak**
…sRowOn in StTpcRSMaker

### CODE REVIEW

File: StRoot/StMcEventMaker/StMcEventMaker.cxx [hunk 1]
```diff
@@ -621,7 +625,7 @@ Int_t StMcEventMaker::Make()
 	    mtdHitTable = g2t_mtd_hitTablePointer->GetTable();
 	else
 	    if (Debug()) cerr << "Table g2t_mtd_hit Not found in Dataset " << geantDstI.Pwd()->GetName() << endl;
-
+#if 0
```
Comment (veprbl) [inline StRoot/StMcEventMaker/StMcEventMaker.cxx]: Is this a left-over from some debugging?

**Comment by plexoos** [comment]
@zhux97 Can you explain what the problem was?

**Comment by zhux97** [comment]
Hi Dmitri,

The detail of problems can be found in these slides
https://drupal.star.bnl.gov/STAR/system/files/embedding_status_20220113.pdf
I think this pull request will address the issues in slide #2, #3, #7, and #10. In particular, the gating grid simulation issue in slide #7.
It would be great if it is merged very soon, and is propagated back to SL21c/SL21d, so that we can start the embedding test.

Thanks!
Xianglei

**Comment by plexoos** [comment]
Thank you for pointing out the slides Xianglei.
Do you think it would make sense to run your tests *before* we merge these changes into the main branch and back port them to SL21c/SL21d?  Can you use a docker or singularity image for the tests? If so, it can be created quickly from the docker tarball generated by the CI jobs for this PR. See the artifacts here: https://github.com/star-bnl/star-sw/actions/runs/1705436926
Please let us know if you need instructions how to use the already built libraries with the proposed changes.

**Comment by zhux97** [comment]
Hi Dmitri,

It will be great to test this PR with a docker. Yes, I do need the instructions.

Thanks,
Xianglei

**Comment by plexoos** [comment]
1. Download the tarball https://github.com/star-bnl/star-sw/suites/4955322829/artifacts/144014736
You need to be logged in to access it or use a github token and a curl command such as:

```
curl -L -u "$GH_USER:$GH_TOKEN"  https://api.github.com/repos/star-bnl/star-sw/actions/artifacts/144014736/zip | tar -xz -C .
```

2. Create an image from the archive file
```
docker load --input star-sw-root5-build.tar
```

3. Run your bfc macro
```
docker run --rm ghcr.io/star-bnl/star-sw-root5-build root4star -b -l -q 'bfc.C(...)'
```

If you need a singularity image, I am sure @veprbl can help

**Comment by zhux97** [comment]
Hi Dmitri,

Thanks! I get the tar ball on SDCC, but the there is no 'docker' command in nodes like rcas6001 & rcas6002. So how I can get this docker running?
Or I had tried singularity image before, and successfully submitted embedding jobs with singularity, so if @veprbl can help on preparing a singularity image, that would be perfect.

Thanks,
Xianglei

**Comment by veprbl** [comment]
You can build a singularity container from a docker tarball using something like `singularity build image.sif docker-archive://star-sw-root6-build.tar`. You will need to download the tar file from github and transfer it to SDCC first.

**Comment by zhux97** [comment]
Thanks! Dmitry, I will try it.

Anyway, I have grabbed the updated code in Yuri's repository, and tested it with DEV at SDCC, the results can be found in
https://drupal.star.bnl.gov/STAR/system/files/embedding_20220119.pdf
In short, the Gating Grid issue and the RDOmap/Mask issue is resolved as expected. But the StMcEventMaker still have some problem (see page #3).

File: StRoot/StMcEventMaker/StMcEventMaker.cxx [hunk 2]
```diff
@@ -1067,12 +1069,14 @@ Int_t StMcEventMaker::Make()
 		long nPseudoPadrow = 0;
 		long ihit;
 		for(ihit=0; ihit<NHits; ihit++) {
-		    if (tpcHitTable[ihit].volume_id < 101 || tpcHitTable[ihit].volume_id > 2445) {
-			if (tpcHitTable[ihit].volume_id <= 202445 &&
-			    tpcHitTable[ihit].volume_id > 2445) nPseudoPadrow++; 
+#if 0 /* keep all tpc hits => Inner TPC sector upgrade */
```
Comment (zhux97) [inline StRoot/StMcEventMaker/StMcEventMaker.cxx:1072]: The line 1072 should be changed to 
#if 1

Comment (plexoos) [reply StRoot/StMcEventMaker/StMcEventMaker.cxx:1072]: Changing it to `#if 1` is equivalent to removing `#if 0` and `#endif` completely

Comment (starsdong) [reply StRoot/StMcEventMaker/StMcEventMaker.cxx:1072]: Hi Yuri/Xianglei, shall we resolve this line and try to merge this request in asap? We also need the commit to patch the SL21c library as well (I see Xianglei submitted a new request, need to be sure they are consistent).

THanks

Comment (zhux97) [reply StRoot/StMcEventMaker/StMcEventMaker.cxx:1072]: In PR289, this line has been changed to 
#if 1
The reason is shown in slide# 3 of
https://drupal.star.bnl.gov/STAR/system/files/embedding_20220119.pdf

Comment (starsdong) [reply StRoot/StMcEventMaker/StMcEventMaker.cxx:1072]: I removed this line from Yuri's branch. If no further comment, I would suggest we merge this into main.

**Comment by plexoos** [comment]
Xin confirmed privately that this is ready to be merged. So, merging it without explicit approvals.

**Pull request merged**